File: task-03/SubTask4/diamond_pattern_file_io.hs
import System.IO

main :: IO ()
main = do
    contents <- readFile "input.txt"
    let n = read (head (lines contents)) :: Int
    writeFile "output.txt" (unlines (diamond n))

diamond :: Int -> [String]
diamond n = [replicate (n-i-1) ' ' ++ replicate (2*i+1) '*' | i <- [0..n-1]]
            ++ [replicate (n-i-1) ' ' ++ replicate (2*i+1) '*' | i <- [n-2,n-3..0]]

File: task-03/SubTask4/diamond_pattern_file_io.go
package main

import (
    "fmt"
    "io/ioutil"
    "strings"
)

func main() {
    content, _ := ioutil.ReadFile("input.txt")
    var n int
    fmt.Sscanf(string(content), "%d", &n)

    var output strings.Builder
    for i := 0; i < n; i++ {
        output.WriteString(strings.Repeat(" ", n-i-1))
        output.WriteString(strings.Repeat("*", 2*i+1))
        output.WriteString("\n")
    }
    for i := n - 2; i >= 0; i-- {
        output.WriteString(strings.Repeat(" ", n-i-1))
        output.WriteString(strings.Repeat("*", 2*i+1))
        output.WriteString("\n")
    }

    ioutil.WriteFile("output.txt", []byte(output.String()), 0644)
}
